pkg/code-generator/codegen/templates/snapshot_template.go

```diff
@@ -80,6 +83,85 @@ func (s {{ .GoName }}Snapshot) HashFields() []zap.Field {
 	return append(fields, zap.Uint64("snapshotHash",  snapshotHash))
 }
 
+func (s *{{ .GoName }}Snapshot) GetResourcesList(resource resources.Resource) (resources.ResourceList, error) {
+	switch resource.(type) {
+{{- range .Resources }}
+	case *{{ .ImportPrefix }}{{ .Name }}:
+		return s.{{ upper_camel .PluralName }}.AsResources(), nil
+{{- end }}
+	default:
+		return resources.ResourceList{}, eris.New("did not contain the input resource type returning empty list")
+	}
+}
+
+func (s *{{ .GoName }}Snapshot) RemoveFromResourceList(resource resources.Resource) error {
+	refKey := resource.GetMetadata().Ref().Key()
+	switch resource.(type) {
+{{- range .Resources }}
+	case *{{ .ImportPrefix }}{{ .Name }}:
+		newList := {{ .ImportPrefix }}{{ .Name }}List{}
+		for _, res := range s.{{ upper_camel .PluralName }} {
+			if refKey != res.GetMetadata().Ref().Key() {
+				newList = append(newList, res)
+			}
+		}
+		s.{{ upper_camel .PluralName }} = newList
+		s.{{ upper_camel .PluralName }}.Sort()
+		return nil	
+{{- end }}
+	default:
+		return eris.Errorf("did not remove the reousource because its type does not exist [%T]", resource)
+	}
+}
+
+func (s *{{ .GoName }}Snapshot) AddOrReplaceToResourceList(resource resources.Resource) error {
```

Comment:
nit; i think upsert is slightly shorter / more readable than `AddOrReplace`, but if this is a painful refactor or others disagree i don't feel super strongly about this

Reply:
agreed upsert is better